lib/rucio/client/commands/rule_commands.py

```diff
@@ -0,0 +1,159 @@
+    """Add replication rule to define how replicas of a list of DIDs are created on RSEs."""
+    args = Arguments(
+        {
+            "dids": dids,
+            "copies": copies,
+            "rse_expression": rses,
+            "weight": weight,
+            "lifetime": lifetime,
+            "grouping": grouping,
+            "locked": locked,
+            "notify": notify,
+            "activity": activity,
+            "comment": comment,
+            "ask_approval": ask_approval,
+            "delay_injection": delay_injection,
+            "rule_account": account,
+            "source_replica_expression": source_rses,
+            "ignore_duplicate": skip_duplicates,
+            "asynchronous": asynchronous,
+        }
+    )
+    add_rule(args, ctx.obj.client, ctx.obj.logger, ctx.obj.console, ctx.obj.spinner)
+
+
+@rule.command("remove")
+@click.argument("rule-id-dids")
+@click.option("--purge-replicas/--no-purge-replicas", default=False, help="Purge rule replicas")
+@click.option("--all/--not-all", "_all", default=False, help="Delete all the rules, even the ones that are not owned by the account")
+@click.option("--rses", "--rse-exp", help="The RSE expression. Must be specified if a DID is provided.")  # TODO mutual inclusive group
+@click.option("--account", help="The account of the rule that must be deleted")
+@click_decorator
+def remove(ctx, rule_id_dids, _all, rses, account, purge_replicas):
+    """Remove an existing rule. Supply [rule-id] if know, or use [DID] and --rses to remove all rules for DIDs on RSEs matching the expression"""
+    args = Arguments({"purge_replicas": purge_replicas, "delete_all": _all, "rule_account": account, "rule_id": rule_id_dids, "rses": rses})
+    delete_rule(args, ctx.obj.client, ctx.obj.logger, ctx.obj.console, ctx.obj.spinner)
+
+
+@rule.command("show")
+@click.argument("rule-id")
+@click.option("--examine/--no-examine", default=False, help="Detailed analysis of transfer errors")
+@click_decorator
+def show(ctx, rule_id, examine):
+    """Retrieve information about a rule"""
+    info_rule(Arguments({"rule_id": rule_id, "examine": examine}), ctx.obj.client, ctx.obj.logger, ctx.obj.console, ctx.obj.spinner)
+
+
+@rule.command("history")
+@click.argument("did", nargs=1)
+@click_decorator
+def history(ctx, did):
+    """Display the history of rules acting on a DID"""
+    list_rules_history(Arguments({"did": did}), ctx.obj.client, ctx.obj.logger, ctx.obj.console, ctx.obj.spinner)
+
+
+@rule.command("move")
+@click.argument("rule_id")
+@click.option("--rses", "--rse-exp", help="RSE expression of new rule", required=True)
+@click.option("--activity", help="Update activity for moved rule", hidden=True)  # Should only do this using `update`
+@click.option("--source-rses", help="Update how replicas are sourced for the rule")
+@click_decorator
+def move(ctx, rule_id, rses, activity, source_rses):
+    """Duplicate a rule onto another RSE"""  # TODO: verify - does it delete the old one?
```

Comment:
No, it doesn’t duplicate. The two rules exist in parallel with a parent-child relationship. When the child rule reaches `OK` state, the parent is deleted.